api/src/database/helpers/fn/dialects/mssql.ts

```diff
@@ -35,7 +35,8 @@ export class FnHelperMSSQL extends FnHelper {
 	}
 
 	count(table: string, column: string, options?: FnHelperOptions): Knex.Raw<any> {
-		const type = this.schema.collections?.[table]?.fields?.[column]?.type ?? 'unknown';
+		const type =
+			this.schema.collections?.[options?.originalCollectionName || table]?.fields?.[column]?.type ?? 'unknown';
```

Comment:
```suggestion
                const collectionName = options?.originalCollectionName || table;
                const type = this.schema.collections?.[collectionName]?.fields?.[column]?.type ?? 'unknown';
```
Perhaps we can split this up for better readability.